### PR TITLE
Enhancement: Support of OSA for zVM HCP

### PR DESCRIPTION
**Pull request by veera-damisetti**
Support of OSA for zVM HCP

- Added changes to support for OSA for zVM HCP
- Changed variable name from osa to interface , to be generic.
- Updated the documentation for the same.

### CODE REVIEW

File: inventories/default/group_vars/all.yaml.template [hunk 1]
```diff
@@ -330,11 +332,11 @@ hypershift:
...
-        - name:
-          host:
-          user:
-          password:
-          osa:
+        - name: 
+          host: 
+          user: 
+          password: 
+          interface:
```
Comment (AmadeusPodvratnik) [inline inventories/default/group_vars/all.yaml.template:339]: Due to the extended scope I would suggest to make a slightly scope. the mentioned id is more or less znet and contains nettype, subchannels and options. Your id variable only contain the subchannels and later you add hardcoded values like qeth or layer2=1. Nettype could be qeth or ctc so I suggest to open this parameter and describe it accordingly.

File: roles/boot_zvm_nodes_hypershift/tasks/main.yaml [hunk 2]
```diff
@@ -43,5 +44,11 @@
...
...oved":true,"hostname":"compute-{{item}}.{{ hypershift.hcp.hosted_cluster_name }}.{{ hypershift.hcp.basedomain }}","installerArgs":"[\"--append-karg\",\"rd.neednet=1\", \"--append-karg\", \"ip={{ hypershift.agents_parms.zvm_parameters.nodes[item].interface.ip }}::{{ hypershift.agents_parms.zvm_parameters.gateway }}:{{ hypershift.agents_parms.zvm_parameters.subnetmask }}:compute-{{ item }}.{{ hypershift.hcp.hosted_cluster_name }}.{{ hypershift.hcp.basedomain }}:{{ hypershift.agents_parms.zvm_parameters.nodes[item].interface.ifname }}:none\", \"--append-karg\", \"nameserver={{ hypershift.agents_parms.zvm_parameters.nameserver }}\", \"--append-karg\",\"rd.znet=qeth,{{ hypershift.agents_parms.zvm_parameters.nodes[item].interface.id }},layer2=1\",\"--append-karg\", {% if hypershift.agents_parms.zvm_parameters.disk_type | lower != 'fcp' %}\"rd.dasd=0.0.{{ hypershift.agents_parms.zvm_parameters.nodes[item].dasd.disk_id }}\"{% else %}\"rd.zfcp={{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].paths[0].fcp}},{{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].paths[0].wwpn }},{{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].id }}\"{% endif %}]"}}' --type merge
```
Comment (AmadeusPodvratnik) [inline roles/boot_zvm_nodes_hypershift/tasks/main.yaml:51]: As mentioned in the all.yaml section there should not be a hardcoded value for nettype or options.

File: roles/setup_for_agents_hypershift/templates/parm-file.parm.j2 [hunk 3]
```diff
@@ -1 +1 @@
...
+rd.neednet=1 console=ttysclp0 coreos.live.rootfs_url=http://{{ hypershift.bastion_hypershift }}:8080/rootfs.img ip={{ hypershift.agents_parms.zvm_parameters.nodes[item].interface.ip }}::{{ hypershift.agents_parms.zvm_parameters.gateway }}:{{ hypershift.agents_parms.zvm_parameters.subnetmask }}::{{ hypershift.agents_parms.zvm_parameters.nodes[item].interface.ifname }}:none nameserver={{ hypershift.agents_parms.zvm_parameters.nameserver }} zfcp.allow_lun_scan=0 rd.znet=qeth,{{ hypershift.agents_parms.zvm_parameters.nodes[item].interface.id }},layer2=1 {% if hypershift.agents_parms.zvm_parameters.disk_type | lower != 'fcp' %}rd.dasd=0.0.{{ hypershift.agents_parms.zvm_parameters.nodes[item].dasd.disk_id }}{% else %}rd.zfcp={{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].paths[0].fcp}},{{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].paths[0].wwpn }},{{ hypershift.agents_parms.zvm_parameters.nodes[item].lun[0].id }} {% endif %} random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8 coreos.inst.persistent-kargs="console=tty1 console=ttyS1,115200n8"
```
Comment (AmadeusPodvratnik) [inline roles/setup_for_agents_hypershift/templates/parm-file.parm.j2:1]: As mentioned in the all.yaml section there should not be a hardcoded value for nettype or options.

**Review by AmadeusPodvratnik — CHANGES_REQUESTED** [review]
Pls specify nettype, subchannels and options for network (ZNET).

File: inventories/default/group_vars/all.yaml.template (hunk 1 above)
Comment (veera-damisetti) [reply inventories/default/group_vars/all.yaml.template:339]: Done

File: roles/setup_for_agents_hypershift/templates/parm-file.parm.j2 (hunk 3 above)
Comment (veera-damisetti) [reply roles/setup_for_agents_hypershift/templates/parm-file.parm.j2]: Done

File: roles/boot_zvm_nodes_hypershift/tasks/main.yaml (hunk 2 above)
Comment (veera-damisetti) [reply roles/boot_zvm_nodes_hypershift/tasks/main.yaml]: Done

**Review by AmadeusPodvratnik — COMMENTED** [review]
Hi @veera-damisetti Thx for the changes ... only one minor comment about this:
arser.add_argument("--network", type=str, help="Network mode for zvm nodes Supported modes: OSA, vswitch, ROCE, Hyperssockets", required=True)
You are mentioning ROCE and Hypersockets but in the description you have only OSA and vswitches.

**Comment by veera-damisetti** [comment]
yeah ROCE and Hipersockets should be removed

File: inventories/default/group_vars/all.yaml.template [hunk 4]
```diff
@@ -331,13 +333,15 @@ hypershift:
...
             ifname: encbdf0
-            id: 0.0.bdf0,0.0.bdf1,0.0.bdf2
+            nettype: qeth
```
Comment (AmadeusPodvratnik) [inline inventories/default/group_vars/all.yaml.template:342]: Hi @veera-damisetti, I believe that the values for ifname, id and nettype are not common or default (maybe for nettype but not sure here). In addition can you pls document the variables.

File: inventories/default/group_vars/all.yaml.template [hunk 5]
```diff
@@ -351,13 +355,15 @@ hypershift:
...
+            subchannels: 0.0.bdf0,0.0.bdf1,0.0.bdf2
```
Comment (AmadeusPodvratnik) [inline inventories/default/group_vars/all.yaml.template:365]: see comment above.

**Review by AmadeusPodvratnik — COMMENTED** [review]
Hi @veera-damisetti , pls see my comments ... only minor.

File: inventories/default/group_vars/all.yaml.template (hunk 5 above)
Comment (veera-damisetti) [reply inventories/default/group_vars/all.yaml.template:365]: Done

File: inventories/default/group_vars/all.yaml.template (hunk 4 above)
Comment (veera-damisetti) [reply inventories/default/group_vars/all.yaml.template:342]: Done @AmadeusPodvratnik Updated the documentation

File: docs/set-variables-group-vars.md [hunk 6]
```diff
@@ -248,14 +248,26 @@
...
 **hypershift.agents_parms.zvm_parameters.nameserver** | Nameserver for compute nodes | 192.168.10.1
 **hypershift.agents_parms.zvm_parameters.subnetmask** | Subnet mask for compute nodes | 255.255.255.0
 **hypershift.agents_parms.zvm_parameters.gateway** | Gateway for compute nodes | 192.168.10.1
 **hypershift.agents_parms.zvm_parameters.nodes** | Set of parameters for zvm nodes <br /> Give the details of each zvm node here | 
+**hypershift.agents_parms.zvm_parameters.nodes[0].name** | Name of the zVM guest | m1317002
+**hypershift.agents_parms.zvm_parameters.nodes[0].host** | Host name of the zVM guests <br /> which we use to login 3270 console | boem1317
+**hypershift.agents_parms.zvm_parameters.nodes[0].user** | Username for zVM guests to login | m1317002
+**hypershift.agents_parms.zvm_parameters.nodes[0].password** | password for the zVM guests to login | password
+**hypershift.agents_parms.zvm_parameters.nodes[0].interface.ifname** | Network interface name for zVM guests | encbdf0
+**hypershift.agents_parms.zvm_parameters.nodes[0].interface.nettype** | Network type for zVM guests for network connectivity | qeth
```
Comment (AmadeusPodvratnik) [inline docs/set-variables-group-vars.md:264]: There is a typo ... the vars should be without [0] ...

**Review by AmadeusPodvratnik — COMMENTED** [review]
Pls remove the typo...

File: docs/set-variables-group-vars.md (hunk 6 above)
Comment (veera-damisetti) [reply docs/set-variables-group-vars.md]: Done

**Review by AmadeusPodvratnik — APPROVED** [review]
@veera-damisetti Looks good. Thank you.

**Pull request merged**